Dockerfile

```diff
@@ -48,16 +51,37 @@ ENV CHPL_HOME=/opt/chapel \
     CHPL_GMP=system \
     CHPL_LLVM=system
 
-# acquire sources
 WORKDIR $CHPL_HOME
+
+
+# Build stage: build Chapel from sources
+# ======================================
+
+FROM chapel-base as chapel-build
+
+# acquire sources
 COPY . .
 
 # build Chapel for both C and LLVM backends
-RUN CHPL_TARGET_COMPILER=llvm make \
-    && CHPL_TARGET_COMPILER=gnu make \
-    && make chpldoc test-venv mason \
-    && make cleanall
+RUN CHPL_TARGET_COMPILER=llvm make -j10 \
+    && CHPL_TARGET_COMPILER=gnu make -j10
+RUN make chpldoc test-venv mason -j10
+RUN make chapel-py-venv chplcheck chpl-language-server -j10
+RUN make cleanall -j10
 
 # Hack to get access to Chapel binaries
 RUN cd $CHPL_HOME/bin && ln -s */* .
+
+# The .git folder is huge and we really don't need it.
+RUN rm -rf .git
+RUN for subdir in `ls test`; do; if [ "$subdir" != "release" ]; then; rm -rf "test/$subdir"; fi; done
```

Comment:
My test build of this finally got here and I get this error:
```
 => ERROR [chapel-build 8/9] RUN for subdir in `ls test`; do; if [ "$subdir" != "release" ]; then;   0.3s
------
 > [chapel-build 8/9] RUN for subdir in `ls test`; do; if [ "$subdir" != "release" ]; then; rm -rf "test/$subdir"; fi; done:
0.283 /bin/sh: 1: Syntax error: ";" unexpected
------
Dockerfile:77
--------------------
  75 |     # The .git folder is huge and we really don't need it.
  76 |     RUN rm -rf .git
  77 | >>> RUN for subdir in `ls test`; do; if [ "$subdir" != "release" ]; then; rm -rf "test/$subdir"; fi; done
  78 |     RUN rm -rf third-party/llvm/llvm-src
  79 |
--------------------
```

I think it's the semicolon on `do;`